From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Geoff Lang <geofflang@chromium.org>
Date: Tue, 9 Jun 2026 14:43:46 -0400
Subject: Request robust access on native EGL contexts.

We would request reset notifications but not robust buffer access.

Covered by existing tests, I haven't seen a driver yet that appears to
test differently by enabling this flag.

Fixed: chromium:516887576
Change-Id: Ie0e931da44199026822b2b27c97f5d34b5d9853d
Reviewed-on: https://chromium-review.googlesource.com/c/angle/angle/+/7914803
Commit-Queue: Geoff Lang <geofflang@chromium.org>
Reviewed-by: Shahbaz Youssefi <syoussefi@chromium.org>

diff --git a/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp b/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp
index 646f6ca16c943dfa80b3c8bc0a5d93216ab166a1..1e26292b219f5a46874cd8fde03797cb79dc247f 100644
--- a/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp
+++ b/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp
@@ -194,6 +194,7 @@ egl::Error DisplayEGL::initializeContext(EGLContext shareContext,
 
             attribsWithRobustness.insert(EGL_CONTEXT_OPENGL_RESET_NOTIFICATION_STRATEGY,
                                          EGL_LOSE_CONTEXT_ON_RESET);
+            attribsWithRobustness.insert(EGL_CONTEXT_OPENGL_ROBUST_ACCESS_EXT, EGL_TRUE);
             if (mHasNVRobustnessVideoMemoryPurge)
             {
                 attribsWithRobustness.insert(EGL_GENERATE_RESET_ON_VIDEO_MEMORY_PURGE_NV, GL_TRUE);
diff --git a/src/tests/angle_end2end_tests_expectations.txt b/src/tests/angle_end2end_tests_expectations.txt
index 748581888416657949b61771959ead7cfe616950..b0ca3422298a7f4b3facc1b7eb0e74967623099e 100644
--- a/src/tests/angle_end2end_tests_expectations.txt
+++ b/src/tests/angle_end2end_tests_expectations.txt
@@ -2203,6 +2203,11 @@
 // Pixel4's GLES driver fails new test: TransformFeedbackTest.CaptureThenUseAsUBO.
 419403192 PIXEL4ORXL GLES : TransformFeedbackTest.CaptureThenUseAsUBO/* = SKIP
 
+// Pixel4 GLES driver fails to compile the GLES1 uber shader when robustness is enabled
+516887576 PIXEL4ORXL GLES : LightsTest.DiffuseGradient/* = SKIP
+516887576 PIXEL4ORXL GLES : LightsTest.FlatLitMesh/* = SKIP
+516887576 PIXEL4ORXL GLES : LightsTest.SmoothLitMesh/* = SKIP
+
 454067880 D3D9 : GLSLTest.ComplicatedMultielementSwizzle1/* = SKIP
 454067880 D3D11 : GLSLTest.ComplicatedMultielementSwizzle1/* = SKIP
 
